### ISSUE TITLE
Bluetooth Connection just once possible.

### ISSUE THREAD

**Issue by bawx25** [Arkq/bluez-alsa#445]
I tried to install AAC. And did it with following steps.

`autoreconf -–install -–force `
then make the direction
`mkdir build`

then adding to the source list with this line
`deb http://www.deb-multimedia.org buster main non-free`

and give the permissions to install it.
`sudo apt-key adv --keyserver keyring.debian.org --recv-keys
5C808C2B65558117`

Followed by a update and upgrade

Now its possible to install libfdk-aac-dev package .

After it i configured the AAC Codec in the build folder with following command. 
`../configure --enable-debug --enable-aac --disable-payloadcheck`

Finally i did
`make
make install`

Everything goes well an I could connect to raspberry pi, i also could play Musik with the aac codec. But after canceling the pairing with the raspi, its not possible to reconnect with the device again. It always says password is incorrect please try again. The option to add a password was never shown so i just press pair und the error appears.

I used this configurations for bluetooth. 

https://github.com/nicokaiser/rpi-audio-receiver/blob/main/install-bluetooth.sh

and after it i installed aac as explained above.

When i watch

`sudo systemctl status bluetooth`

than there is this error

May 21 12:33:28 Ramplify systemd[1]: Started Bluetooth service.
May 21 12:33:28 Ramplify bluetoothd[555]: Starting SDP server
May 21 12:33:28 Ramplify bluetoothd[555]: Bluetooth management interface 1.18 initialized
May 21 12:33:28 Ramplify bluetoothd[555]: Sap driver initialization failed.
May 21 12:33:28 Ramplify bluetoothd[555]: sap-server: Operation not permitted (1)
May 21 12:33:28 Ramplify bluetoothd[555]: Endpoint registered: sender=:1.3 path=/org/bluez/hci0/A2DP/AAC/Sink/1
May 21 12:33:28 Ramplify bluetoothd[555]: Failed to set privacy: Rejected (0x0b)
May 21 12:33:28 Ramplify bluetoothd[555]: Endpoint registered: sender=:1.3 path=/org/bluez/hci0/A2DP/AAC/Sink/2
May 21 12:33:28 Ramplify bluetoothd[555]: Endpoint registered: sender=:1.3 path=/org/bluez/hci0/A2DP/SBC/Sink/1
May 21 12:33:28 Ramplify bluetoothd[555]: Endpoint registered: sender=:1.3 path=/org/bluez/hci0/A2DP/SBC/Sink/2

and the `/etc/systemd/system/bluetooth.target.wants/bluetooth.service` this

ExecStart=/usr/lib/bluetooth/bluetoothd --compat --noplugin=sap -E

and for the privacy error i used this https://github.com/RPi-Distro/pi-bluetooth/issues/8#issuecomment-480270680.

The Errors go away but the reconnection doesnt work. 

You can download my Image under this
link: https://studhsheilbronnde-my.sharepoint.com/:u:/g/personal/tbreunig_stud_hs-heilbronn_de/EQoDuo5i0eBFsDY7r76xjnABnVzb8QaPYl249akuK-jECQ?e=sOsi2O 

After all i noticed when i run a bash skript which delete all ever connected bluetooth devices i can connect again. 

#!/bin/bash
for device in $(bt-device -l | grep -o "[[:xdigit:]:]\{11,17\}"); do
    echo "removing bluetooth device: $device | $(bt-device -r $device)"
done

**Comment by borine** [comment]
> But after canceling the pairing with the raspi

Please can you explain what you mean by that? How do you "cancel the pairing" ? Why do you ever need to "cancel" a pairing?

This looks like an issue with whatever bluetooth management application you are using on the Pi. I always use the command-line `bluetoothctl`application to pair, trust and connect my audio devices, and have no such issues. 

You should pair with a device only once, to set up the encryption keys. This info is then permanently stored on the device and the Pi so there is normally no need to do it again. After pairing, it is necessary to "connect" the device when you want to use it - connection can happen automatically or require manual intervention depending on how you have set up the system.

**Comment by bawx25** [comment]
With canceling the connection I mean to disconnect the smartphone completly. You need it sometimes because when you accidently delete the connection you cant connect back and need to install everything again. Also i didnt pair the Smartphone with the `bluetoothctl` i can connect direktly without any screen. My raspi shows me the opertunity to connect and i just connect.

**Comment by borine** [comment]
"cancelling the connection" is not the same thing as "cancelling the pairing". As I said above, you pair only once (usually, but not always, if pairing a mobile phone you will be asked to enter a security check number). 

Then you need to connect the phone to the Pi. It is usual to tell the Pi to "trust" the phone and then connections are initiated from the 'phone; the Pi should accept the connection request because it "trusts" the phone. Does your bluetooth management app always set the "trust" attribute for all paired devices or is there a separate dialog for you to enable that? You can see if the device is trusted by running the command:
```
bluetoothctl info XX:XX:XX:XX:XX:XX
```
where XX:XX:XX:XX:XX:XX is the bluetooth address of the 'phone.

**Comment by borine** [comment]
> accidently delete the connection

hmm, perhaps you mean that you accidentally removed the *pairing* on the phone - in that case the pairing info stored on the Pi is no longer valid, and that might explain why you are getting the unexpected "password incorrect" message. In which case, you need to delete the pairing info on the Pi first, and then go back to the start and pair the phone with the Pi again. That will create new, matching, credentials on both devices and then you should be able to connect again.

I delete pairing info on my Pi by doing:
```
bluetoothctl remove XX:XX:XX:XX:XX:XX
```
No need to remove any other devices, just the 'phone where the "accident" happened.

**Comment by borine** [comment]
This issue appears to be about the use of the RPi bluetooth management applet rather than bluez-alsa. Given that RaspiOS has now switched to pulseaudio for use with their desktop, and that this issue has seen no activity for a few weeks, I am closing it as no longer relevant. If new information becomes available then it can be re-opened.
